modules/api/src/main/java/org/apache/ignite/compute/JobExecutionContext.java

```diff
@@ -36,4 +38,13 @@ public interface JobExecutionContext {
      * @return {@code true} when the job was cancelled.
      */
     boolean isCancelled();
+
+    /**
+     * List of partitions numbers associated with this job. Not a {@code null} only when
+     * {@link IgniteCompute#submitBroadcastPartitioned(String, JobDescriptor, Object)} method is used to submit jobs. In this case, the list
+     * contains partitions numbers that are local on the node executing the job.
+     *
+     * @return list of partitions numbers associated with this job.
+     */
+    @Nullable List<Integer> partitions();
```

Comment:
```suggestion
    @Nullable List<org.apache.ignite.table.partition.Partition> partitions();
```

Should be compatible with `PartitionManager` API.